httprequest_lego_provider/forms.py

```diff
@@ -2,8 +2,66 @@
 # See LICENSE file for licensing details.
 """Forms."""
 
+import re
+
+from django.core.exceptions import ValidationError
 from django.forms import CharField, Form
 
+FQDN_PREFIX = "_acme-challenge."
+
+
+def _is_fqdn(fqdn) -> bool:
+    """Check if the argument is a valid FQDN.
+
+    Args:
+        fqdn: the FQDN to validate.
+
+    Returns:
+        if the FQDN is valid.
+    """
+    return bool(
+        re.match(
+            (
+                r"^(?!.{255}|.{253}[^.])([a-z0-9](?:[-a-z-0-9]{0,61}[a-z0-9])?\.)+"
+                r"[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?[.]?$"
+            ),
+            fqdn,
+            re.IGNORECASE,
+        )
+    )
+
+
+def is_fqdn_compliant(fqdn) -> bool:
```

Comment:
I think an input type annotation here can help :D

Reply:
Will add in the next PR. Sorry